File: src/main/java/com/pairingplanet/pairing_planet/domain/entity/post/recipe/RecipeStep.java
package com.pairingplanet.pairing_planet.domain.entity.post.recipe;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "recipe_steps")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class RecipeStep {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Long postId;
    private Integer version;

    @Column(name = "step_number", nullable = false)
    private Integer stepNumber;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String description;

    @Column(name = "image_url")
    private String imageUrl;
}

File: lib/domain/entities/recipe/recipe_step.dart
class RecipeStep {
  final int stepNumber;
  final String? description;
  final String? imageUrl;
  final String? imagePublicId;

  RecipeStep({
    required this.stepNumber,
    required this.description,
    this.imageUrl,
    this.imagePublicId,
  });
}
